lib/src/barback/dartdevc/module_computer.dart

```diff
@@ -353,6 +354,26 @@ class _ModuleComputer {
     return modulesById.values.toList();
   }
 
+  /// Renames shared [Module]s to something unique and short to avoid issues
+  /// with file names that are too long.
+  List<Module> _renameSharedModules(List<Module> modules) {
+    if (modules.isEmpty) return modules;
+    var next = 0;
+    // All modules and assets in those modules share a top level dir, we just
+    // grab it from the first one.
```

Comment:
As a part of one of the other pull requests I added a `dir` field to `ModuleId` so I don't have to rely on this semi-hack any more, see https://github.com/dart-lang/pub/pull/1580/commits/4d3302ad5d01109d4ec36cf0feb1dfde50172a3b.